### PR TITLE
Do not merge nor copy monitoring tables when merging DL1 h5 files

### LINKED ISSUE #1193: Different length of `.../monitoring/flatfield` in subruns blocks DL1 merging
While merging DL1 files, OSA failed for one of them with the following error:

```python
Traceback (most recent call last):
  File "/fefs/aswg/software/conda/envs/osa-v0.10/bin/lstchain_merge_hdf5_files", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/fefs/aswg/software/conda/envs/osa-v0.10/lib/python3.11/site-packages/lstchain/scripts/lstchain_merge_hdf5_files.py", line 117, in main
    auto_merge_h5files(
  File "/fefs/aswg/software/conda/envs/osa-v0.10/lib/python3.11/site-packages/lstchain/io/io.py", line 362, in auto_merge_h5files
    raise ValueError("Length of key {} from file {} different than in file {}".format(k, filename, file_list[0]))
ValueError: Length of key /dl1/event/telescope/monitoring/flatfield from file /fefs/aswg/data/real/DL1/20231211/v0.10/tailcut84/dl1_LST-1.Run15982.0052.h5 different than in file /fefs/aswg/data/real/DL1/20231211/v0.10/tailcut84/dl1_LST-1.Run15982.0000.h5
```
The merged DL1 file was halfway produced, though.

For approximately half of the files of run `dl1_LST-1.Run15982.*.h5`  the length of the flatfield table is 1, while for the other half is 2. For other runs is always 2. 

Is this expected @FrancaCassol?

### PR DESCRIPTION
**Pull request by morcuended**
closes #1193

### CODE REVIEW

File: lstchain/io/io.py [hunk 1]
```diff
@@ -353,14 +353,6 @@ def auto_merge_h5files(
...
 
             with open_file(filename) as file:
-
-                # check value of Table.nrow for keys copied from the first file
-                for k in keys_to_copy:
-                    first_node = merge_file.root[k]
-                    present_node = file.root[k]
-                    if first_node.nrows != present_node.nrows:
-                        raise ValueError("Length of key {} from file {} different than in file {}".format(k, filename, file_list[0]))
-
```
Comment (moralejo) [inline lstchain/io/io.py:363]: I don't understand, this is not specific of the monitoring table... won't this change have unwanted consequences for other tables?

Comment (morcuended) [reply lstchain/io/io.py:363]: `keys_to_copy`currently includes by default:

```
default_keys_to_copy = [
    dl1_params_tel_mon_ped_key, 
    dl1_params_tel_mon_cal_key,
    dl1_params_tel_mon_flat_key,
    dl1_mon_tel_CatB_cal_key,
    dl1_mon_tel_CatB_ped_key,
    dl1_mon_tel_CatB_flat_key
]
```
so all of them are copied from the first subrun and then checked that the length of these tables is the same for all subruns which is not true at least for `dl1_params_tel_mon_flat_key` as reported in #1193

Shall I then remove the first three tables from the `keys_to_copy`? They will be merged then

**Comment by FrancaCassol** [comment]
Hi, agree with the change, for the moment I would just remove the check of the length. 
In a short future the second line of (dl1_params_tel_mon_ped_key, dl1_params_tel_mon_cal_key, dl1_params_tel_mon_flat_key) will be in anyway suppressed since we are going to use 
(dl1_mon_tel_CatB_cal_key, dl1_mon_tel_CatB_ped_key, dl1_mon_tel_CatB_flat_key)

**Comment by morcuended** [comment]
> Can you make that they are neither merged nor copied?

I brought back the check of the same length, but I removed the monitoring tables from the keys to be copied list. Also, remove those keys from the ones to be merged as suggested by @moralejo.

File: lstchain/scripts/lstchain_merge_hdf5_files.py [hunk 2]
```diff
@@ -23,20 +23,15 @@
...
-default_keys_to_copy = [dl1_params_tel_mon_ped_key, 
-                       dl1_params_tel_mon_cal_key,
-                       dl1_params_tel_mon_flat_key,
-                       dl1_mon_tel_catB_cal_key,
-                       dl1_mon_tel_catB_ped_key,
-                       dl1_mon_tel_catB_flat_key
+default_keys_to_copy = [
+    dl1_mon_tel_catB_cal_key,
+    dl1_mon_tel_catB_ped_key,
+    dl1_mon_tel_catB_flat_key
```
Comment (moralejo) [inline lstchain/scripts/lstchain_merge_hdf5_files.py:34]: Are these safe from  having issues like those with the old monitoring tables? Do they have one entry per subrun?

Comment (moralejo) [reply lstchain/scripts/lstchain_merge_hdf5_files.py:34]: The new version has been tested, right? Changes look ok to me, but one never knows...

Comment (morcuended) [reply lstchain/scripts/lstchain_merge_hdf5_files.py:34]: > The new version has been tested on data and MC, right? Changes look ok to me, but one never knows...

Yes, merging is tested on both data and MC. I wanted to wait for #1257 to get the coverage reports back

**Review by moralejo — APPROVED** [review]

**Pull request merged**